### DISCUSSION TITLE
Maintaining Socket Connection in Mobile Sleep Mode - MERN Stack

### DISCUSSION THREAD

**Discussion by soomyProd** [socketio/socket.io#4892] [Q&A]
**Question:**

I'm working on a MERN stack application where I'm utilizing Socket.IO for real-time communication. Everything works well, but I'm facing an issue when the mobile device goes into sleep mode. The socket connection seems to be getting disrupted during this state (disconnected).

**Context:**

I have a backend setup using Node.js and Socket.IO, and on the frontend side, I'm using React with the useEffect hook to establish a socket connection when the component mounts. Here's a simplified version of my code:

**Backend (socket.js):**

```js
// ... (previous code)
io.on("connect", (socket) => {
  socket.on("join", async () => {
    // some code...
  } catch (error) {
    // handle error...
  }
});
```
**Frontend (product.js):**

```js
// ... (previous code)
useEffect(() => {
  socket = io();
  if (/* some condition */) {
    socket.emit("join", {
      // some code...
    });
    // socket open on user join
    return () => {
      socket.disconnect();
    };
  }
}, []);
```

**Issue:**

The socket connection works as expected, but when the mobile device goes to sleep mode, the connection seems to be lost. I've tried implementing a solution, but I'm unsure about the best approach to keep the socket connection active during sleep mode.

**Request for Help:**

 1. What are the recommended strategies or best practices to maintain a
    socket connection when a mobile device goes into sleep mode?
 2. Are there specific configurations or events I should be aware of in
    Socket.IO to handle such scenarios?
 3. Any insights or code examples related to handling socket connections
    on mobile devices, especially in the context of mobile web applications?

Any assistance or guidance would be greatly appreciated. Thanks in advance!

**Comment by vuggy17** [comment]
Same problem when an edge tab goes sleep (on desktop), there is no disconnect event fired on my server. And using server.fetchSocket() doesn't include the sleep tab.

My solution for this is to maintain a heartbeat mechanism to check whether the tab is online.

@soomyProd did you found any other solution?
